backend/capellacollab/projects/toolmodels/routes.py

```diff
@@ -125,26 +127,69 @@ def patch_tool_model(
         injectables.get_existing_capella_model
     ),
     db: orm.Session = fastapi.Depends(database.get_db),
+    user: users_models.DatabaseUser = fastapi.Depends(
+        users_injectables.get_own_user
+    ),
 ) -> models.DatabaseCapellaModel:
-    version = get_version_by_id_or_raise(db, body.version_id)
-    if version.tool != model.tool:
-        raise fastapi.HTTPException(
-            status_code=status.HTTP_409_CONFLICT,
-            detail={
-                "reason": f"The tool having the version “{version.name}” (“{version.tool.name}”) does not match the tool of the model “{model.name}” (“{model.tool.name}”)."
-            },
-        )
+    if body.version_id:
+        version = get_version_by_id_or_raise(db, body.version_id)
+        if version.tool != model.tool:
+            raise fastapi.HTTPException(
+                status_code=status.HTTP_409_CONFLICT,
+                detail={
+                    "reason": f"The tool having the version “{version.name}” (“{version.tool.name}”) does not match the tool of the model “{model.name}” (“{model.tool.name}”)."
+                },
+            )
+    else:
+        version = model.version
+
+    if body.nature_id:
+        nature = get_nature_by_id_or_raise(db, body.nature_id)
+        if nature.tool != model.tool:
+            raise fastapi.HTTPException(
+                status_code=status.HTTP_409_CONFLICT,
+                detail={
+                    "reason": f"The tool having the nature “{nature.name}” (“{nature.tool.name}”) does not match the tool of the model “{model.name}” (“{model.tool.name}”)."
+                },
+            )
+    else:
+        nature = model.nature
```

Comment:
That is true but probably we can just use `if nature and nature.tool != model.tool` to resolve this.
In addition, while looking at my suggestion, I figured out that it might make sense to explicitly state `if body.nature_id is not None` since `0` is a valid id for me and with my current suggestion it would not work properly.